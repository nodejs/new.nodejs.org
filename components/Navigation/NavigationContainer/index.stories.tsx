import NavigationContainer from './';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';
import type { FC } from 'react';

type Story = StoryObj<typeof NavigationContainer>;
type Meta = MetaObj<typeof NavigationContainer>;

const NavigationContent: FC = () => (
  <ul>
    <li>Navigation item 1</li>
    <li>Navigation item 2</li>
    <li>Navigation item 3</li>
  </ul>
);

export const Default: Story = {
  args: {
    isOpen: false,
    label: 'Navigation',
    children: <NavigationContent />,
  },
};

export default { component: NavigationContainer } as Meta;
