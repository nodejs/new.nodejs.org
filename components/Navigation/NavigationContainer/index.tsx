import { FormattedMessage } from 'react-intl';
import classnames from 'classnames';
import styles from './index.module.scss';
import type { FC, PropsWithChildren } from 'react';

type NavigationContainerProps = PropsWithChildren<{
  isOpen: boolean;
  label: string;
  toggleNavigation: () => void;
}>;

const NavigationContainer: FC<NavigationContainerProps> = ({
  label,
  isOpen,
  children,
  toggleNavigation,
}) => {
  const navigationClasses = classnames(styles.navigation, {
    [styles.navigationFixed]: isOpen,
  });

  return (
    <nav aria-label={label} className={navigationClasses} id="main-navigation">
      <button
        type="button"
        className={styles.navigationOpen}
        onClick={toggleNavigation}
        aria-expanded={isOpen}
      >
        <FormattedMessage id="components.navigation.navigationContainer" />
      </button>
      {children}
    </nav>
  );
};

export default NavigationContainer;
