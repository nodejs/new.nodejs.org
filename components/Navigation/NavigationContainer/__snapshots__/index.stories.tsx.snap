// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/NavigationContainer Default smoke-test 1`] = `
<nav aria-label="Navigation"
     class="NavigationContainer_navigation__KY7SY"
     id="main-navigation"
>
  <button type="button"
          class="NavigationContainer_navigationOpen__WdiUj"
          aria-expanded="false"
  >
    Menu
  </button>
  <ul>
    <li>
      Navigation item 1
    </li>
    <li>
      Navigation item 2
    </li>
    <li>
      Navigation item 3
    </li>
  </ul>
</nav>
`;
